### DISCUSSION TITLE
Couldn't connect to the controller

### DISCUSSION THREAD

**Discussion by jazap** [dec0dOS/zero-ui#214] [Q&A]
Stumped on  this. After "docker compose up" I keep getting message:

zu-main        | Couldn't connect to the controller on http://localhost:9993/
zu-main        | Couldn't connect to the controller on http://localhost:9993/

Tried changing docker-compose.yml "ZU_CONTROLLER_ENDPOINT=" to the actual url where I connect to my ztncui and still get the same error.

Any help appreciated.

**Comment by nityam2007** [comment]
change localhost to your public ip 
and expose it over firewall 

I had used it without caddy here is code if u want to use it without caddy and make sure u use public ip , I got error on 0.0.0.0 , 127.0.0.1 , as well on localhost 

here is code without caddy 

version: "3"

services:
  zerotier:
    image: zyclonite/zerotier:1.10.6
    container_name: zu-controller
    restart: unless-stopped
    volumes:
      - ./zerotier-one:/var/lib/zerotier-one
    environment:
      - ZT_OVERRIDE_LOCAL_CONF=true
      - ZT_ALLOW_MANAGEMENT_FROM=0.0.0.0/0
    expose:
      - "9993"
    ports:
      - "9993:9993"
  zero-ui:
    image: dec0dos/zero-ui:latest
    container_name: zu-main
    build:
      context: .
      dockerfile: ./docker/zero-ui/Dockerfile
    restart: unless-stopped
    depends_on:
      - zerotier
    volumes:
      - ./zerotier-one:/var/lib/zerotier-one
      - ./data:/app/backend/data
    environment:
      - ZU_CONTROLLER_ENDPOINT=http://<your_public_ip>:9993/
      - ZU_SECURE_HEADERS=false
      - ZU_DEFAULT_USERNAME=admin
      - ZU_DEFAULT_PASSWORD=admin
    expose:
      - "4000"
    ports:
      - "4000:4000"
